docs/okeps/okep-5193-user-defined-networks.md

```diff
@@ -0,0 +1,1468 @@
+
+Additionally, multiple and different instances of primary networks may co-exist for different users, and they will provide
+native network isolation.
+
+## Terminology
+
+* **Primary Network** - The network which is used as the default gateway for the pod. Typically recognized as the eth0
+  interface in the pod.
+* **Secondary Network** - An additional network and interface presented to the pod. Typically created as an additional
+  Network Attachment Definition (NAD), leveraging Multus. Secondary Network in the context of this document refers to a
+  secondary network provided by the OVN-Kubernetes CNI.
+* **Cluster Default Network** - This is the routed OVN network that pods attach to by default today as their primary network.
+  The pods default route, service access, as well as kubelet probe are all served by the interface (typically eth0) on this network.
+* **User-Defined Network** - A network that may be primary or secondary, but is declared by the user.
+* **Layer 2 Type Network** - An OVN-Kubernetes topology rendered into OVN where pods all connect to the same distributed
+  logical switch (layer 2 segment) which spans all nodes. Uses Geneve overlay.
+* **Layer 3 Type Network** - An OVN-Kubernetes topology rendered into OVN where pods have a per-node logical switch and subnet.
+  Routing is used for pod to pod communication across nodes. This is the network type used by the cluster default network today.
+  Uses Geneve overlay.
+* **Localnet Type Network** - An OVN-Kubernetes topology rendered into OVN where pods connect to a per-node logical switch
+  that is directly wired to the underlay.
+
+## Goals
+
+* Provide a configurable way to indicate that a pod should be connected to a user-defined network of a specific type as a
+  primary interface.
+* The primary network may be configured as a layer 3 or layer 2 type network.
+* Allow networks to have overlapping pod IP address space. This range may not overlap with the default cluster subnet
+  used for allocating pod IPs on the cluster default network today.
+* The cluster default primary network defined today will remain in place as the default network pods attach to. The cluster
+  default network will continue to serve as the primary network for pods in a namespace that has no primary user-defined network. Pods
+  with primary user-defined networks will still attach to the cluster default network with limited access to Kubernetes system resources.
+  Pods with primary user-defined networks will have at least two network interfaces, one connected to the cluster default network and one
+  connected to the user-defined network. Pods with primary user-defined networks will use the user-defined network as their default
+  gateway.
+* Allow multiple namespaces per network.
+* Support cluster ingress/egress traffic for user-defined networks, including secondary networks.
+* Support for ingress/egress features on user-defined primary networks where possible:
+    * EgressQoS
```

Comment:
would you like to remove qos and svc from this enhancement's scope (the presence of ESVC support as part of this enhancement is what caused confusion for our telco friends since we descoped it later)
I think the new QoS API covers the support for multi-networks so this enhancement can remove that one and for ESVC maybe move it into future goals like with MEG?

Reply:
on second thoughts maybe its the feature docs (I am writing it its just not been fast enough should call out these things)

Reply:
I'll move egress service to future goals, and change EgressQoS to just QoS, since it is now supported.